### ISSUE TITLE
File existence tests do not follow symlinks

### ISSUE THREAD

**Issue by bgauduch** [GoogleContainerTools/container-structure-test#182]
Hello 👋 

I am using `fileExistenceTest` to validate Apache 2 module are present on an alpine image, but it looks like it doesn't follow symlinks.
On Alpine, Apache modules are installed in `/usr/lib/apache2` directory, and symlinked to `/var/www/modules` by default.

### Dockerfile:
Simplified Dockerfile used to build the Image with Kaniko:
```dockerfile
FROM alpine:3.8
RUN apk add --no-cache  apache2=2.4.35-r0
```

### Apache 2 module configuration:
```bash
# stripped down to only 1 module for clarity
LoadModule macro_module modules/mod_macro.so
```

### Test module presence with `apache_file_tests.yaml`:
```yaml
schemaVersion: "2.0.0"

fileExistenceTests:
  - name: 'check module macro is installed'
    path: '/var/www/modules/mod_macro.so'
    shouldExist: true
```

### Structure test command:
```bash
container-structure-test test --driver tar --image my-apache:latest --config apache_file_tests.yaml
```

### Expected behavior is Test: PASS, but I get:
```bash
INFO: File Existence Test: check module macro is installed
time="2018-11-14T11:54:27Z" level=info msg="Removing image filesystem directory /tmp/git.int.rcacloud.it:4567dockerapache@sha256:089d7cdd6ec79105ae70225a344f7d760eae700af647665adb1bd0ff484a2f0f280365698 from system"
=== RUN: File Existence Test: check module macro is installed
--- FAIL
Error: Error examining file in container: stat /tmp/git.int.rcacloud.it:4567dockerapache@sha256:089d7cdd6ec79105ae70225a344f7d760eae700af647665adb1bd0ff484a2f0f280365698/var/www/modules/mod_macro.so: no such file or directory
```

### Test without symlink pass without issue:
```yaml
schemaVersion: "2.0.0"

fileExistenceTests:
  - name: 'check module macro is installed'
    path: '/usr/lib/apache2/mod_macro.so'
    shouldExist: true
```
```bash
INFO: File Existence Test: check module macro is installed
time="2018-11-14T11:54:26Z" level=info msg="Removing image filesystem directory /tmp/git.int.rcacloud.it:4567dockerapache@sha256:089d7cdd6ec79105ae70225a344f7d760eae700af647665adb1bd0ff484a2f0f276330479 from system"
=== RUN: File Existence Test: check module macro is installed
--- PASS
```

### Content of folder `/var/www` inside running container
Symlinks are present as expected inside running container from the Dockerfile.
```bash
# ls -la /var/www/
total 12
drwxr-xr-x    3 root     root          4096 Nov 14 11:32 .
drwxr-xr-x    1 root     root          4096 Nov 14 11:32 ..
drwxr-xr-x    4 root     root          4096 Nov 14 11:32 localhost
lrwxrwxrwx    1 root     root            16 Nov 14 11:32 logs -> /var/log/apache2
lrwxrwxrwx    1 root     root            16 Nov 14 11:32 modules -> /usr/lib/apache2
lrwxrwxrwx    1 root     root            12 Nov 14 11:32 run -> /run/apache2
```

### Additional informations
When exploring the temporary filesystem used by structure test (using `--save` flag and a temporary volume mount), the symlinks are not present.

Edit : 
- inverted Apache module location in test scripts.

**Comment by nkubala** [comment]
Hey @bgauduch, thanks for the issue. Have you tried to reproduce this using an image not build with Kaniko? I'm wondering if the underlying issue is in Kaniko's handling of symlinks, or if it's an issue here with cst. I haven't had a chance to try and repro yet, but I'll try and get to it soon.

**Comment by bgauduch** [comment]
Hello @nkubala,

Thank's for the feedback 😄

I have just tested again:
- building the image with Kaniko and running the exact same test : it fail.
- building the image with Docker and running the exact same test : it works.

So it clearly seem's to be related to the way Kaniko handle symlinks 😢

Besides, we (my current project) have had similar problems since the opening of this issues, so we have set Kaniko aside to use "standard" docker build instead (yes, privileged mode needed, but it's more stable).

I believe this issue can be closed, we might try again in the future when other build system (Kaniko and other) will be more stable).

**Comment by nkubala** [comment]
@bgauduch thanks for testing it out again. Really sorry to hear that Kaniko has given you issues, symlinks in particular have been a tough issue for us to solve. We'll take a look at this in Kaniko and see if we can figure it out, but feel free to open an issue for this there, and also for any other bugs you run into.

Definitely give Kaniko another shot in the near future - we've been working really hard to get it stable, and with the new caching system enabled we've been seeing builds happen faster on average than Docker!

**Comment by priyawadhwa** [comment]
hey @bgauduch super sorry to hear that kaniko isn't working for you -- if you've got some spare time, it would be great if you could open an [issue](https://github.com/GoogleContainerTools/kaniko/issues) in the kaniko repo! We'll try to take a look ASAP and see if we can fix this symlink issue :)

**Comment by bgauduch** [comment]
Hey guys,

Thanks again for your time and feedbacks.
We will most certainly use Kaniko again in the future, depending of the use-case of course 😉 

I have opened a [new issue](https://github.com/GoogleContainerTools/kaniko/issues/481) on the Kaniko repo.
We have already burned a lot of fuel trying to push Kaniko in our CI, so it is quite terse as we obviously have to move on. Besides there seem's to be a lot of issues related to symlinks already.

Hope you guys we go over thoose symlinks issues 👍 

Regards,
Baptiste
